### PR TITLE
[Jobs] Extending existing logs

### PR DESCRIPTION
**Pull request by fhibf**
* Extending existing logs

## FHIR Team Checklist
- [x] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Operations/JobLoggerExtension.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.SqlServer.Features.Operations;
+using Microsoft.Health.JobManagement;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Operations
```
Comment (mikaelweave) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Operations/JobLoggerExtension.cs:14]: Could this be in a common area where CosmosDB jobs could use it too?

**Review by mikaelweave — COMMENTED** [review]
One specific question with the code and one general question - when querying these logs in the future will we be able to query by a column (like JobId) in the `Logs` table or will this information be in the message?

File: src/Microsoft.Health.Fhir.SqlServer/Features/Operations/JobLoggerExtension.cs (hunk 1 above)
Comment (fhibf) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Operations/JobLoggerExtension.cs:14]: That would be applicable for the old jobs, is that the intension?

Comment (mikaelweave) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Operations/JobLoggerExtension.cs:14]: Here is an example on the *new* job framework that occurs on CosmosDB. 

https://github.com/microsoft/fhir-server/blob/53f5153126e604573441c61a82b02d713b9c6ebe/src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/Export/CosmosExportOrchestratorJob.cs#L25

**Review by mikaelweave — APPROVED** [review]

**Pull request merged**